### DISCUSSION TITLE
Dendron Frontend Backlog

### DISCUSSION THREAD

**Discussion by kevinslin** [dendronhq/dendron#2491] [General]
## Summary
This thread started as a conversation between me and @aleksey-rowan on current work items with the Dendron frontend. 

I've listed some details about frontend initiatives we're currently thinking about - if anyone is interested in helping out with any of these items, feel free to post in this thread with the specific item and I'm happy to add additional details and isntructions

## Dendron Frontend Stack

- react
- redux
- typescript
- antd design framework 
- less (as a consequence of using antd)

### Support Custom Themes
- this will be similar to how markdown preview enhanced does it: https://shd101wyy.github.io/markdown-preview-enhanced/#/customize-css
- allow users to use custom css in both preview and in publishing

### Support Current VSCode Theme
- vscode exposes current theme properties in the webview
- dendron should use these properties when loading its own theme 

### Unified Styling
- Dendron's current styles is a composition of 
    - https://pmarsceill.github.io/just-the-docs/
    - https://mmistakes.github.io/minimal-mistakes/
    - https://ant.design/docs/spec/introduce

There's a lot of overlap in the stylesheets, some of it with conflicts

### Mobile Styling Issues

There are currently multiple issues with how certain elements display on mobile (eg. when using lookup, pressing enter does not collapse lookup results)

### Typora like view for dendron
- built a custom editor that supports inline styles much like the [markless](https://github.com/tejasvi/markless) extension

**Comment by ghost** [comment]
> Support Current VSCode Theme
I think should come as a first step for theming, since it would only require wiring up the VS-Code Variables to the needed CSS Classes that get set by `antd` 

and then we already have this wiring for 

> Support Custom Themes

and then we need less work to support custom themes, at least for everything that is already supported through the compatibility with the VSCode Theme
